packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1675,6 +1674,15 @@ export class ContainerRuntime
 			throw error;
 		}
 
+		// DuplicateBatchDetection is only enabled if Offline Load is enabled
+		// It maintains a cache of all batchIds/sequenceNumbers within the collab window.
+		// Don't waste resources doing so if not needed.
+		this.duplicateBatchDetector = this.offlineEnabled
+			? new DuplicateBatchDetector()
+			: new (class NoOpDuplicateBatchDetector implements IDuplicateBatchDetector {
```

Comment:
can this just undefined if not enabled?

if not, you don't need to create a class, just create and object with the function
```
{
processInboundBatch : () => ({ duplicate: false as const });
}
```
its less code, and less memory